Lab2_optimized.kt

```diff
@@ -0,0 +1,52 @@
+/* Programador: Fernanda Esquivel (esq21542@uvg.edu.gt).
+ * Lenguaje: Kotlin
+ * Recursos: Kotlin Playground
+ * Historial: Finalizado el 24.07.2022 */
+
+// No tocar esta clase ---
+data class ItemData(
+    var originalPos: Int,
+    var originalValue: Any,
+	var type: String? = null,
+    var info: String? = null
+)
+// -----------------------
+
+fun main() {
+    val result = processList(listOf(25, "Hola", null, false))
+    result?.forEach{ println(it) }
+}
+
+fun processList(inputList: List<Any?>?): List<ItemData>? {
+    var returnList: MutableList<ItemData>? = mutableListOf<ItemData>(); //Lista de retorno
+    if (inputList == null) { return null }
+    else { inputList.forEach { 
+        	when (it) { is Int -> { val item = ItemInt(it, inputList.indexOf(it)); returnList?.add(item) }
```

Comment:
No es muy recomendable utilizar `indexOf(it)` pues puede llegarle a consumir mucha más memoria y fallar en algunos escenarios para el ejercicio.